google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeployToolsMenuAction.java

```diff
@@ -69,4 +80,39 @@ public void actionPerformed(AnActionEvent event) {
       }
     }
   }
+
+  /**
+   * Determines if the project has at least one module with an App Engine standard or flexible
+   * facet. If it does not, then a notification balloon is shown.
+   */
+  @VisibleForTesting
+  boolean isAppEngineProjectCheck(@NotNull Project project) {
```

Comment:
Could the first part of this method, the part that does the actual check to see if a project is an App Engine project be moved to a utility function?